plugins/attachment-resources/src/components/AttachmentStyleBoxEditor.svelte

```diff
@@ -27,27 +27,27 @@
   export let key: KeyedAttribute
   export let placeholder: IntlString
   export let focusIndex = -1
-  let _id: Ref<Doc> | undefined = undefined
-  let _class: Ref<Class<Doc>> | undefined = undefined
   const client = getClient()
 
   const queryClient = createQuery()
 
-  let description = ''
+  let description: string
+  let haveUnsavedChanges = false
 
-  let doc: Doc | undefined
-  function checkForNewObject (object: Doc) {
-    if (object._id !== _id) return true
-    if (object._class !== _class) return true
-    return false
+  $: if (object && description === undefined) {
+    description = getAttribute(client, object, key)
   }
+
+  // We need to query the document one more time
+  // To make a difference between update of description from the bottom
+  // And update to if from another tab.
   $: object &&
-    queryClient.query(object._class, { _id: object._id }, async (result) => {
-      ;[doc] = result
-      if (doc && checkForNewObject(object)) {
-        _class = object._class
-        _id = object._id
-        description = getAttribute(client, object, key)
+    queryClient.query(object._class, { _id: object._id }, async (result: Doc[]) => {
+      if (result.length > 0) {
+        if (!haveUnsavedChanges) {
```

Comment:
Actually we could try do some magic here, collect changes user did and try apply them to document he edited.